### PR TITLE
managarm-system: use `mold` instead of `ld`

### PR DESCRIPTION
**Pull request by no92**

### CODE REVIEW

**Comment by no92** [comment]
Depends on managarm/managarm#683.

**Comment by ArsenArsen** [comment]
(not against the change.. in fact, i have already messed with mold before, just curious)

what's the reason for this change?

**Comment by ElectrodeYT** [comment]
> (not against the change.. in fact, i have already messed with mold before, just curious)
> 
> what's the reason for this change?

It's quite a bit faster, and reduces iteration time by a decent amount, especially on slower machines, if a change is made that requires relinking several servers or other binaries.

File: bootstrap.d/sys-devel.yml [hunk 1]
```diff
@@ -61,6 +61,26 @@ tools:
     install:
       - args: ['make', 'install']
 
+  - name: host-mold
```
Comment (Dennisbonke) [inline bootstrap.d/sys-devel.yml:64]: I miss the label tag (if applicable) and the architecture flag. Can you add them?

**Review by Dennisbonke — CHANGES_REQUESTED** [review]
One issue

**Comment by no92** [comment]
Fixed the issue and rebased onto master.

**Review by Dennisbonke — APPROVED** [review]
LGTM

**Pull request merged**